### PR TITLE
samples: net: lwm2m: use real temperature values instead of dummy

### PR DESCRIPTION
**Pull request by therealprof**
This changes the use of a static and fixed temperature value to using a
read_callback which will either return the previous dummy value or
gather real temperature data from a supported sensor (for now only
FXOS8700 which is available e.g. on a FRDM-K64F).

Signed-off-by: Daniel Egger <daniel@eggers-club.de>

### CODE REVIEW

**Comment by therealprof** [comment]
Not sure how to fix the shippable error. `IS_ENABLED` is preferred to checking #defines but for this still barfs for DT symbols. Of course I could also combine both and define the DT label if it is not defined but yuck...

**Comment by jukkar** [comment]
> `IS_ENABLED` is preferred to checking #defines but for this still barfs for DT symbols. Of course I could also combine both and define the DT label if it is not defined but yuck...

IS_ENABLED() is indeed preferred but in some cases is not possible to do it like that. Just do
```
#if defined(CONFIG_FXOS8700_TEMP)
		dev = device_get_binding(DT_NXP_FXOS8700_0_LABEL);
#endif 
```
in this case.

**Comment by mike-scott** [comment]
@therealprof love the change -- I need a few days before my time opens up to review (and 1.14 LTS rolls out)

In our sources we make all of the HW temp device names the same via these settings:
https://github.com/foundriesio/dm-lwm2m/blob/master/Kconfig#L72
https://github.com/foundriesio/dm-lwm2m/blob/master/boards/frdm_k64f.overlay#L11
and then generically use them:
https://github.com/foundriesio/dm-lwm2m/blob/master/src/main.c#L28
https://github.com/foundriesio/dm-lwm2m/blob/master/src/main.c#L82

**Comment by therealprof** [comment]
> In our sources we make all of the HW temp device names the same via these settings:

So you're creating custom labels using the device tree and use them in the application? I'm not sure that approach really scales for general samples in Zephyr as we probably don't want to create a separate overlay per potential hardware device which could be used which also this gets very messy if a user attaches an external sensor to a generic board.

**Comment by mike-scott** [comment]
Haven't forgotten about this!  Just got back from a small Spring vacation, clearing out the work items from 1.14 and I'll get to this soon (along with some new client changes as well).

**Comment by therealprof** [comment]
@mike-scott Gentle monthly ping 😅

File: samples/net/lwm2m_client/src/lwm2m-client.c [hunk 1]
```diff
@@ -177,6 +178,32 @@ static int firmware_update_cb(u16_t obj_inst_id)
...
+	dev = device_get_binding(DT_NXP_FXOS8700_0_LABEL);
+#endif
+
+	if (dev != NULL) {
+		if (sensor_sample_fetch(dev)) {
+			LOG_ERR("temperature data update failed");
+		}
+
+		sensor_channel_get(dev, SENSOR_CHAN_DIE_TEMP,
+				(struct sensor_value *) &v);
```
Comment (mike-scott) [inline samples/net/lwm2m_client/src/lwm2m-client.c]: Let's get this rebased on today's master and fix 1 minor nit: this line needs to be aligned with the parenthesis above it like so (added 3 spaces):
```
		sensor_channel_get(dev, SENSOR_CHAN_DIE_TEMP,
				   (struct sensor_value *) &v);
```

Sorry this took so long :/  work has been.. odd.

**Review by mike-scott — CHANGES_REQUESTED** [review]

File: samples/net/lwm2m_client/src/lwm2m-client.c (hunk 1 above)
Comment (therealprof) [reply samples/net/lwm2m_client/src/lwm2m-client.c]: Done.

**Review by mike-scott — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**